### DISCUSSION TITLE
v0.9.0 is not available for Mac M1 chip

### DISCUSSION THREAD

**Discussion by itsAmitGoyani** [SeaQL/sea-orm#880] [Q&A] [answered]
## Description
sea-orm version 0.9.0 is not available for Mac M1 chip.
Error:
rust-analyzer failed to load workspace: Failed to read Cargo metadata from Cargo.toml file api/rust/Cargo.toml, cargo 1.59.0: Failed to run `"cargo" "metadata" "--format-version" "1" "--manifest-path" "/api/rust/Cargo.toml" "--filter-platform" "aarch64-apple-darwin"`: `cargo metadata` exited with an error: Blocking waiting for file lock on package cache Updating crates.io index error: failed to select a version for the requirement `sea-orm = "^0.9"` candidate versions found which didn't match: 0.8.0, 0.7.1, 0.7.0, ... location searched: crates.io index required by package `apis v0.1.0 (/api/rust/apis)`

Output of `cargo check`:
    Updating crates.io index
error: failed to select a version for the requirement `sea-orm = "^0.9"`
candidate versions found which didn't match: 0.8.0, 0.7.1, 0.7.0, ...
location searched: crates.io index
required by package `apis v0.1.0 (/api/rust/apis)`

## Additional Information

Please refer to attached screenshots for more information.
![Screenshot 2022-07-15 at 11 13 11 AM](https://user-images.githubusercontent.com/63532626/179159409-552d35db-0d85-4398-9a4d-d60fecce97a9.png)

**Comment by baoyachi** [comment, marked as answer]
Try update local rust version to latest .

```bash
~ rustup update
~ rustc -V
~ rustc 1.62.0 (a8314ef7d 2022-06-27)
```

Then exec cmd: `cargo b`

**Comment by billy1624** [comment]
Thanks for reporting!! @itsAmitGoyani

**Comment by billy1624** [comment]
Please let me know if upgrading Rust version helps or not

Thanks! @baoyachi

**Comment by baoyachi** [comment]
It's seems cargo bug.

**Comment by baoyachi** [comment]
@billy1624 This caused by link ref: https://github.com/rust-lang/cargo/issues/10688

**Comment by itsAmitGoyani** [comment]
@baoyachi @billy1624 Thanks a lot. My `rustup` version was 1.62.0 only. Just problem was my `rustc` path was set to separately installed location of version 1.59.0. But now I uninstalled it and it worked. Thanks again.

**Comment by billy1624** [comment]
Interesting! Thanks all :)

**Comment by tyt2y3** [comment]
Interesting. I have a M1 Mac too so I sometimes would build SeaORM on it.
